### ISSUE TITLE
Plantuml parse error

### ISSUE THREAD

**Issue by cheoljoo** [doxygen/doxygen#6473]
# environment
- macbook
  - $  uname -a
  - Darwin cheoljoolees-MacBook-Pro.local 17.3.0 Darwin Kernel Version 17.3.0: Thu Nov  9 18:09:22 PST 2017; root:xnu-4570.31.3~1/RELEASE_X86_64 x86_64
  - doxygen 1.8.15  has the following error. (developing version)
- linux ubuntu 16.04
  - doxygen 1.8.14 did not have error.  the result was good.
  - doxygen 1.8.15 has the following error. 

# Issue
- I attached 2 files. One (fail.cpp) has the following error msg and the other (succeed.cpp) is success.
- difference between two files.
  - Just add * in starting point of line

## Error Message
```
/Users/cheoljoo/Code/LG/test_doxygen/fail.cpp:12: warning: explicit link request to 'FFFFFF' could not be resolved
/Users/cheoljoo/Code/LG/test_doxygen/fail.cpp:13: warning: explicit link request to 'ADD1B2' could not be resolved
/Users/cheoljoo/Code/LG/test_doxygen/fail.cpp:14: warning: explicit link request to 'FFFFFF' could not be resolved
/Users/cheoljoo/Code/LG/test_doxygen/fail.cpp:21: warning: unexpected command enduml
```

## Attachment
The file with the `fail.cpp` and the `succeed.cpp`: [err.zip](https://github.com/doxygen/doxygen/files/2334649/err.zip)

**Comment by albert-github** [comment]
The problem lies in the way the file is parsed and the usage of the markdown parser. The markdown parser moved its place for consistency in the 1.8.15 version.

 - it is a bit strange to have 2 types of comment style mixed (with `*` and without).

__Workarounds:__

 - disable `MARKDOWN_SUPPORT` but this will, probably, impact results in other parts of the output as well
 - add, in this case, 1 space in front of all lines without the * in the comment. In fact the add some many spaces as there are before the `*` + 1 (so here 0 + 1)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #6491

**Comment by albert-github** [comment]
I don't think the previous proposed patch #6491 is correct (also might change other parts).
Replaced proposed patch by proposed patch [#10999](https://github.com/doxygen/doxygen/pull/10999)

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
